### PR TITLE
Added support for Clojure programming language

### PR DESCRIPTION
**Pull request by RohanVashisht1234**

### CODE REVIEW

File: plugins/language_clojure.lua [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  files = {
+    "%.clojure$",
+  },
+  patterns = {
+    { pattern = ';;.*',                                   type = 'comment' },
+    { pattern = ';.*',                                    type = 'comment' },
+    { pattern = { '#"', '"', '\\' },                      type = 'string' },
+    { pattern = { '"', '"', '\\' },                       type = 'string' },
+    { pattern = { '"""', '"""', '\\' },                   type = 'string' },
+    { pattern = 'Retention%s+()[%a_][%w_/]*',             type = { 'keyword', 'literal' } },
```
Comment (Guldoman) [inline plugins/language_clojure.lua:16]:
```suggestion
    { pattern = 'Retention()%s+()[%a_][%w_/]*',             type = { 'keyword', 'normal', 'literal' } },
```

File: plugins/language_clojure.lua [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    { pattern = ':[%a_][%w_/%-]*',                        type = { 'keyword2', 'literal' } },
```
Comment (Guldoman) [inline plugins/language_clojure.lua:17]: There is no group here, but `type` is a table.

File: plugins/language_clojure.lua [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    { pattern = '[%a_][%w_]*()%.()[%a_][%w_/%-]*',        type = { 'keyword','operator', 'keyword2' } },
+    { pattern = "%(()def()%s+()[%a_][%w_%-]*",            type = { "normal", "keyword", "literal", 'literal' } }, -- tested ok
+    { pattern = "%(()def[%a_][%w_]*()%s+()[%a_][%w_%-]*", type = { "normal", "keyword", "literal", 'literal' } }, -- tested ok
```
Comment (Guldoman) [inline plugins/language_clojure.lua:20]: Is this about things that start with `def`? So even things like `def_whatever`?

File: manifest.json [hunk 4]
```diff
@@ -735,6 +736,16 @@
       ],
       "version": "0.2"
     },
+    {
+      "description": "Syntax for the [Clojure programming language](https://clojure.org/).",
```
Comment (Guldoman) [inline manifest.json:740]:
```suggestion
      "description": "Syntax for the [Clojure](https://clojure.org/) programming language",
```

File: plugins/language_clojure.lua (hunk 3 above)
Comment (RohanVashisht1234) [reply plugins/language_clojure.lua:20]: yes, anything that starts with def is to be highlighted

File: plugins/language_clojure.lua (hunk 2 above)
Comment (RohanVashisht1234) [reply plugins/language_clojure.lua]: fixed

**Comment by Guldoman** [comment]
Seems like clojure files mostly use a different file extension than what's being checked here.
Would you mind making another PR adding those?
For example in the LSP repository we use these: https://github.com/lite-xl/lite-xl-lsp/blob/59602bc153b567026acc0822fd535a6d82748ce9/config.lua#L132

**Comment by RohanVashisht1234** [comment]
Yes, I have fixed that

**Pull request merged**